### PR TITLE
[14.0][FIX] stock_available_to_promise_release: priority propagation on release

### PR DESCRIPTION
**Pull request by TDu**
The management of priorities on stock moves and stock picking has changed
between on v14 see

* https://github.com/odoo/odoo/commit/e40553723b893bcd2244e9b9fcc5c1a7e1bf27d3

This make the priority field that was nicely flowing between stock move
through the procurement values, not working any more.

That change on Odoo core omits the inverse function on the priority
field in stock move, but it would not help when going through the
procurement mechanism, because moves are created before they are
assigned to picking.

### CODE REVIEW

File: stock_available_to_promise_release/models/stock_move.py [hunk 1]
```diff
@@ -324,6 +324,7 @@ def _run_stock_rule(self):
                 backorder_links[new_move.picking_id] = move.picking_id
 
             values = move._prepare_procurement_values()
+
```
Comment (sebalix) [inline stock_available_to_promise_release/models/stock_move.py]: New line to remove (to reduce the diff)

File: stock_available_to_promise_release/models/stock_move.py [hunk 2]
```diff
@@ -403,3 +409,10 @@ def _release_split(self, remaining_qty):
         new_move._assign_picking()
 
         return new_move.with_context(context)
+
+    def _assign_picking_post_process(self, new=False):
+        super()._assign_picking_post_process(new)
+        highest_priorities = self.mapped("move_dest_ids.picking_id.priority")
+        if highest_priorities:
+            highest_priorities.sort()
+            self.picking_id.write({"priority": highest_priorities[-1]})
```
Comment (mmequignon) [inline stock_available_to_promise_release/models/stock_move.py]:
```suggestion
            self.picking_id.write({"priority": max(highest_priorities)})
```
I suppose… If you agree, you can apply the same everywhere else.

**Review by mmequignon — APPROVED** [review]

File: stock_available_to_promise_release/models/stock_move.py (hunk 2 above)
Comment (mmequignon) [reply stock_available_to_promise_release/models/stock_move.py]:
```python
>>> max(["1", "12", "42", "0"])
'42'
```

File: stock_available_to_promise_release/models/stock_move.py (hunk 1 above)
Comment (TDu) [reply stock_available_to_promise_release/models/stock_move.py]: Removed

File: stock_available_to_promise_release/models/stock_move.py (hunk 2 above)
Comment (TDu) [reply stock_available_to_promise_release/models/stock_move.py]: Thanks, I updated the pr

File: stock_available_to_promise_release/models/stock_move.py [hunk 3]
```diff
@@ -371,6 +371,10 @@ def _after_release_update_chain(self):
             moves = moves.mapped("move_orig_ids")
         pickings = self.env["stock.picking"].browse(picking_ids)
         pickings._after_release_update_chain()
+        # Set the highest priority on all pickings in the chain
+        highest_priorities = pickings.mapped("priority")
```
Comment (sebalix) [inline stock_available_to_promise_release/models/stock_move.py]:
```suggestion
        priorities = pickings.mapped("priority")
```
The highest is then given by the call to `max`, not before :p

Comment (TDu) [reply stock_available_to_promise_release/models/stock_move.py]: True, renamed.

Comment (sebalix) [reply stock_available_to_promise_release/models/stock_move.py]: It remains one place where it should be renamed ;)

Comment (TDu) [reply stock_available_to_promise_release/models/stock_move.py]: Should be good now.

**Comment by simahawk** [comment]
@sebalix good for you?

**Review by sebalix — APPROVED** [review]
LG, would be good to have this in a dedicated method but not blocking.

**Pull request merged**